src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -2519,6 +2519,34 @@ static inline int icap_xmc_free(struct icap *icap)
 	return err == -ENODEV ? 0 : err;
 }
 
+static bool check_group_topo_and_connectivity(struct icap *icap,
+	struct axlf *xclbin)
+{
+    struct connectivity *group_conn = icap->connectivity;
+	struct mem_topology *group_topo = icap->mem_topo;
+	const struct axlf_section_header *hdr = get_axlf_section_hdr(icap, xclbin, ASK_GROUP_TOPOLOGY);
+
+	if (!hdr || !group_topo)
+		return false;
+
+    /* If group_topology section presents, overwrite mem_topology with it */
+    memcpy(group_topo, ((const char *)xclbin) + hdr->m_sectionOffset,
```

Comment:
Did you make sure the buffer pointed to by group_topo is big enough for you to copy a new section to it? I'm not sure I understand what you are trying to do here... After the copy, we don't see original mem topo section from the xclbin any more in driver? I thought that we have discussed this before and already made it clear that mem topo is going to be used by xbutil. You can use mem topo section as group topo, if group topo is not found. But you can't use group topo as mem topo, right? Same thing below for connectivity section...